packages/vscode-extension/src/debugging/ProxyDebugAdapter.ts

```diff
@@ -148,67 +148,48 @@ export class ProxyDebugAdapter extends DebugSession {
   ) {
     await this.cdpProxy.initializeServer();
 
-    let didStartSessionHandler: Disposable | null = vscode.debug.onDidStartDebugSession(
-      (session) => {
-        if (session.type === CHILD_SESSION_TYPE) {
-          this.nodeDebugSession = session;
-          didStartSessionHandler?.dispose();
-          didStartSessionHandler = null;
-        }
+    const debugSession = await startDebugging(
+      undefined,
+      {
+        type: CHILD_SESSION_TYPE,
+        name: "Radon IDE Debugger",
+        request: "attach",
+        port: this.cdpProxy.port!,
+        continueOnAttach: true,
+        sourceMapPathOverrides: args.sourceMapPathOverrides,
+        resolveSourceMapLocations: ["**", "!**/node_modules/!(expo)/**"],
+        skipFiles: [
+          "**/extension/lib/**/*.js",
+          "**/vscode-extension/lib/**/*.js",
+          "**/ReactFabric-dev.js",
+          "**/ReactNativeRenderer-dev.js",
+          "**/node_modules/**/*",
+          "!**/node_modules/expo-router/**/*",
+        ],
+      },
+      {
+        suppressDebugStatusbar: true,
+        suppressDebugView: true,
+        suppressDebugToolbar: true,
+        suppressSaveBeforeStart: true,
+        parentSession: this.session,
+        consoleMode: vscode.DebugConsoleMode.MergeWithParent,
+        lifecycleManagedByParent: true,
+        compact: true,
       }
     );
 
-    try {
-      const childSessionStarted = await debug.startDebugging(
+    if (!debugSession) {
```

Comment:
If I read `startDebugging` correctly, it always returns the session, so `debugSession` is always truthy here.
You should wrap the `startDebugging` call in a `try/catch` instead.